[flang][driver] support -dumpversion and -dumpmachine

yuanfang-chen opened the pull request:
Match GCC driver. GCC has -cc1/-fc1 support too, but this patch does not address that.

banach-space commented on clang/lib/Driver/Driver.cpp:
```diff
@@ -2081,7 +2082,7 @@ bool Driver::HandleImmediateArgs(const Compilation &C) {
     // Since -dumpversion is only implemented for pedantic GCC compatibility, we
     // return an answer which matches our definition of __VERSION__.
-    llvm::outs() << CLANG_VERSION_STRING << "\n";
+    llvm::outs() << DriverVersion << "\n";
```
This change is not needed, is it?

banach-space reviewed (commented):
Could this be implemented without any updates to Clang (beyond Options.td)?

yuanfang-chen commented:
> Could this be implemented without any updates to Clang (beyond Options.td)?

The options could work by changing options.td only. However,  `CLANG_VERSION_STRING` and `FLANG_VERSION_STRING` could be different. This is to handle that.

banach-space commented:
> > Could this be implemented without any updates to Clang (beyond Options.td)?
> 
> The options could work by changing options.td only. However, `CLANG_VERSION_STRING` and `FLANG_VERSION_STRING` could be different. This is to handle that.

You are adding a support for these new flags in `flang-new`, which is implemented in terms of `clangDriver` - this scenario would be very unlikely at the moment. And even if that was desired, you'd need to update Flang's driver too. I'd rather keep this to the required minimum. If you really want to keep this, you will need to update flang/tools/flang-driver/driver.cpp too.

yuanfang-chen committed:
[flang][driver] support -dumpversion and -dumpmachine

Match GCC driver. GCC has -cc1/-fc1 support too, but this patch does not address that.

yuanfang-chen commented:
> You are adding a support for these new flags in `flang-new`, which is implemented in terms of `clangDriver` - this scenario (i.e. the two projects being out of sync) would be very unlikely at the moment (perhaps even impossible). And even if that was desired, you'd need to update Flang's driver too. I'd rather keep this to the required minimum. If you really want to keep this, you will need to update flang/tools/flang-driver/driver.cpp too.

Thanks for the explanation. I've updated the patch.

banach-space reviewed (approved):
LGTM, thanks!
